<?xml version="1.0" encoding="utf-8"?>	

<launch>

    <arg name="simulator" default="/vrep" />
    <arg name="robot_name" default="ugv1" />  <!-- this also defines the robot id, the last digit after "ugv" -->  

    <!-- pioneer robot https://www.generationrobots.com/media/Pioneer3DX-P3DX-RevA.pdf -->    
    <arg name="robot_radius" default = "0.4"/>     

    <arg name="pointcloud_in" default="$(arg simulator)/$(arg robot_name)/local_map_normals"/> 

    <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->
          
    <arg name="enable_multi_robot_avoidance" default="true" />  <!-- multi -->
    <arg name="number_of_robots" default="2"/>                  <!-- multi -->
  

    <node name="traversability_$(arg robot_name)" pkg="path_planner" type="traversability" respawn="$(arg respawn_value)" output="screen">
        
        <rosparam file="$(find  path_planner)/launch/path_planner_octomap.yaml" /> 

        <param name="robot_name" value="$(arg robot_name)" /> 

        <param name="robot_frame_name" value="$(arg robot_name)/base_link"/>   

        <param name="robot_radius" value="$(arg robot_radius)"/> 
        
        <param name="number_of_robots" value="$(arg number_of_robots)" /> 

        <param name="enable_multi_robot_avoidance" value="$(arg enable_multi_robot_avoidance)"/>  <!-- multi -->

        <!-- Input -->
        <remap from="/dynjoinpcl" to="$(arg pointcloud_in)"/>
        <remap from="/obst_point_cloud" to="$(arg simulator)/$(arg robot_name)/obst_point_cloud"/>

        <!-- Output -->
        <remap from="/clustered_pcl/no_wall" to="$(arg simulator)/$(arg robot_name)/clustered_pcl/no_wall"/>
        <remap from="/clustered_pcl/wall" to="$(arg simulator)/$(arg robot_name)/clustered_pcl/wall"/>
        <remap from="/trav/traversability" to="$(arg simulator)/$(arg robot_name)/trav/traversability"/>

        <remap from="/trav/clearence" to="$(arg simulator)/$(arg robot_name)/trav/clearence"/>
        <remap from="/trav/density" to="$(arg simulator)/$(arg robot_name)/trav/density"/>     
        <remap from="/trav/label" to="$(arg simulator)/$(arg robot_name)/trav/label"/>          
        <remap from="/trav/roughness" to="$(arg simulator)/$(arg robot_name)/trav/roughness"/>  

        <remap from="/clustered_pcl/segmented" to="$(arg simulator)/$(arg robot_name)/clustered_pcl/segmented"/>  
        <remap from="/normals_pcl" to="$(arg simulator)/$(arg robot_name)/normals_pcl"/>  
        <remap from="/normals_marker" to="$(arg simulator)/$(arg robot_name)/normals_marker"/>  

    </node>

</launch>
